### DISCUSSION TITLE
DB-embedded for "out of the box" experience

### DISCUSSION THREAD

**Discussion by mrizzi** [guacsec/trustify#33] [Ideas]
Thinking about the user that wants to give the tool a try running easily it on his host: a single executable binary is the optimal result to let the user "download and run".

In this scenario, the DB should be embedded and considering on "production" environments PostgreSQL external instance has been proposed, I've investigated the options for having PostgreSQL embedded instead of others (e.g. SQLite).

I came to this project https://github.com/theseus-rs/postgresql-embedded
It looks maintained.
I gave it a quick test and it looks like it works fine also in regards to full text search feature from PostgreSQL we identified as a solution for delivering search capabilities.
Furthermore it allows to bundle PostgreSQL that in my test increases the final binary size just by ~10 MB.

**Comment by bobmcwhirter** [comment]
I think perhaps an initial foray to evaluating this would be replacing our external pgsql for our integration-y unit tests that presume you have a pgsql running.

**Comment by helio-frota** [comment]
This project is interesting

I also quick tested (this is not the correct location of the setup) to see what happens:

![2024-03-08_16-35](https://github.com/trustification/trustify/assets/6443576/b3252d1f-f3a2-4114-bdf1-e8280bc0d096)

when running:

```
RUST_LOG=debug cargo run --bin trustify-cli importer csaf --full-source-url ../../trustification/data/ds1/csaf --db-user postgres --db-password ""
```

it starts and creates a db, as I have no idea how to config the password it creates a new password every time and a new port is allocated.

```
[2024-03-08T19:31:44Z DEBUG postgresql_embedded::postgresql] Creating database huevos for localhost:38583
[2024-03-08T19:31:44Z DEBUG postgresql_embedded::command::traits] Executing command: PGPASSWORD="5k9dCRktveeJhdvF"
```

not able to connect probably because the random port and password
```
Error: Connection Error: pool timed out while waiting for an open connection
Caused by:
	pool timed out while waiting for an open connection
```

**Comment by bobmcwhirter** [comment]
First attempt in this (failing) PR https://github.com/trustification/trustify/pull/40
